### ISSUE TITLE
update Ingress to apiVersion: networking.k8s.io/v1

### ISSUE THREAD

**Issue by fzhan** [rook/rook#7905]
**Is this a bug report or feature request?**
* Feature Request
update ingress yaml to adept apiVersion: networking.k8s.io/v1
**What should the feature do:**
- current setup doesn't work under k8s 1.20

**What is use case behind this feature:**
- after update to the following code, URL responded with code 503
***Sample Code***
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: rook-ceph-mgr-dashboard
  namespace: rook-ceph # namespace:cluster
  annotations:
    kubernetes.io/tls-acme: "true"
    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
    nginx.ingress.kubernetes.io/server-snippet: |
      proxy_ssl_verify off;
spec:
  ingressClassName: public
  tls:
    - hosts:
        - rook.services.local
      secretName: rook.services.local
  rules:
    - host: rook.services.local
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: rook-ceph-mgr-dashboard
                port:
                  name: https-dashboard
```

**Environment**:
rook-ceph v release-1.6
k8s: 1.20.4

**Comment by travisn** [comment]
@fzhan Rook doesn't create the ingress, Rook just creates the internal service for the dashboard. Are you able to get a response from the dashboard if you connect to the service from inside the cluster? For example, from the toolbox can you curl the service? If so, the dashboard is working as expected and you need to find the ingress settings that will work in your environment.

**Comment by fzhan** [comment]
> @fzhan Rook doesn't create the ingress, Rook just creates the internal service for the dashboard. Are you able to get a response from the dashboard if you connect to the service from inside the cluster? For example, from the toolbox can you curl the service? If so, the dashboard is working as expected and you need to find the ingress settings that will work in your environment.

Thanks for the update, I'm currently using the service from inside the cluster, above ingress configuration came with the example, just thought it'd be best to update to support a newer version of k8s.

**Comment by travisn** [comment]
Oh right, I see now, we have a couple examples with the `apiVersion: extensions/v1beta1` for the Ingress type. Did it work for you if you just changed to `v1`? Or was any other change needed to the example? thanks

**Comment by fzhan** [comment]
@travisn That's what I did, I've also followed the official guideline and added pathType, service name, ingressClass etc. Unfortunately I could not access gist, so the code pasted in the thread looks messy.

**Comment by travisn** [comment]
@fzhan Ok this would be addressed now by #7933. For the formatting, the triple tick ``` does the trick. :)

**Comment by fzhan** [comment]
@travisn brilliant! Thanks~
